extensions/wikia/Chat2/ChatEntryPoint.class.php

```diff
@@ -23,35 +23,56 @@ static public function onParserFirstCallInit( Parser &$parser ) {
 	}
 
 	/**
-	 * @param $isEntryPoint - set to true for chat entry point embeded in the article, false for rail module
-	 * Return an array of variables needed to render chat entry point template
+	 * @param $isEntryPoint - set to true for chat entry point embedded in the article,
+	 * set to false for rail module
+	 *
+	 * Return an array of variables needed to render chat entry point mustache template
 	 *
 	 * @return array
 	 */
 	static public function getEntryPointTemplateVars( $isEntryPoint ) {
-		global $wgEnableWallExt, $wgBlankImgUrl;
-		return [
-			'linkToSpecialChat' => SpecialPage::getTitleFor( "Chat" )->escapeLocalUrl(),
-			'isEntryPoint' => $isEntryPoint,
+		global $wgEnableWallExt, $wgBlankImgUrl, $wgUser, $wgSitename;
+
+		$entryPointGuidelinesMessage = wfMessage( 'chat-entry-point-guidelines' );
+		$joinTheChatMessage = wfMessage( 'chat-join-the-chat' );
+		$chatUsers = ChatEntryPoint::getChatUsersInfo();
+		$chatProfileAvatarUrl = AvatarService::getAvatarUrl( $wgUser->getName(), ChatRailController::AVATAR_SIZE );
+
+		$vars =  [
 			'blankImgUrl' => $wgBlankImgUrl,
-			'profileType' => !empty( $wgEnableWallExt ) ? 'message-wall' : 'talk-page'
+			'chatUsers' => $chatUsers,
+			'chatUsersCount' => count( $chatUsers ),
+			'entryPointGuidelinesMessage' => $entryPointGuidelinesMessage->exists() ?
+				$entryPointGuidelinesMessage->parse() : null,
+			'isEntryPoint' => $isEntryPoint,
```

Comment:
This apparently tells you if the chat widget is rendered in the right rail or if it is included using `<chat/>` parser tag. I can see a big room for improvement to name it better but don't actually have any particular idea. Do you have any?